tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+     * ResolvedMethodDeclaration: JavaParserMethodDeclaration,
+     * ReflectionMethodDeclaration, and JavassistMethodDeclaration. A
+     * signature follows the format:
+     *     "[modifiers] [typeParameters] [type] [methodName]([parameters]) throws [exceptions]"
+     */
+    public static String getMethodSignature(
+            MethodUsage methodUsage
+    ) {
+        ResolvedMethodDeclaration methodDeclaration = methodUsage.getDeclaration();
+        // Consider JavaParserMethodDeclaration.
+        if (methodDeclaration instanceof JavaParserMethodDeclaration jpMethodDeclaration) {
+            MethodDeclaration jpMethod = jpMethodDeclaration.getWrappedNode();
+            return getMethodSignature(jpMethod);
+        }
+        // Consider ReflectionMethodDeclaration or JavassistMethodDeclaration
+        Matcher matcher = METHOD_SIGNATURE.matcher(methodDeclaration.toString());
+        if (!matcher.find()) {
+            throw new IllegalStateException("Could not parse method signature: " + methodDeclaration);
+        }
+        String methodModifiers = methodDeclaration instanceof ReflectionMethodDeclaration ? matcher.group(1) : matcher.group(2);
+        // Take into account the case in which the method declaration refers to a method without an access specifier.
+        if (methodModifiers == null) {
+            methodModifiers = "";
+        }
+        List<String> typeParameterList = getTypeParameters(methodUsage);
+        List<String> formalParameterList = getParameters(methodUsage);
+        List<String> exceptionList = getExceptions(methodUsage);
+        return (methodModifiers + " " + (typeParameterList.isEmpty() ? "" : "<" + String.join(", ", typeParameterList) + ">") +
+                " " + getTypeWithoutPackages(methodDeclaration.getReturnType()) +
+                " " + methodDeclaration.getName() +
+                "(" + String.join(", ", formalParameterList) + ")" +
+                (exceptionList.isEmpty() ? "" : " throws " + String.join(", ", exceptionList)))
+                .replaceAll(" +", " ").trim();
+    }
+
+    /**
+     * Gets the JavaParser package declaration of a given compilation unit.
+     *
+     * @param cu a compilation unit
+     * @return the JavaParser package declaration {@link PackageDeclaration}
+     * of the compilation unit
+     * @throws PackageDeclarationNotFoundException if the JavaParser package
+     * declaration {@link PackageDeclaration} cannot be found or is unnamed
+     */
+    public static PackageDeclaration getPackageDeclaration(
+            CompilationUnit cu
+    ) throws PackageDeclarationNotFoundException {
+        Optional<PackageDeclaration> jpPackage = cu.getPackageDeclaration();
+        if (jpPackage.isEmpty()) {
+            throw new PackageDeclarationNotFoundException("The compilation unit has no package declaration");
+        }
+        return jpPackage.get();
+    }
+
+    /**
+     * Returns the base component type of a resolved type. Recursively strips
```

Comment:
In Java, the "base component type" is called the "element type".
So, the component type of int[][][] is int[][], and the element type of int[][][] is int.